### DISCUSSION TITLE
Nuxt3 set base pages directory on runtime

### DISCUSSION THREAD

**Discussion by larsjanssen6** [nuxt/nuxt#26188] [Questions]
I've got a nuxt3 directory structure like this:

```
pages/app-one/index.vue
pages/app-two/index.vue
```

Right now I set in my `nuxt.config.ts`, the `dir` property to `app-one`, and I see app-one correctly when I visit `/`.

How can I make sure I can determine the app root based on the domain name (on run-time)? I cannot push routes manually because I have a lot of dynamic routes.

**Comment by xjccc** [comment]
Something like this?
```
routeRules: {
  '/': { redirect: '/app-one/' }
}
```

**Comment by larsjanssen6** [comment]
@xjccc thanks for helping. This would be in my nuxt.config file right? I need to determine this step based on the `window.location.hostname`.  Then this would not work I guess.

**Reply by xjccc** [reply]
You could use a global middleware to change?

[middleware](https://nuxt.com/docs/guide/directory-structure/middleware#usage)

**Reply by larsjanssen6** [reply]
That would be more some kind of redirect. I want to set a 'base' directory. From that base directory (in pages) the index.vue should be served from there.

**Reply by xjccc** [reply]
I think your 'base' is decided by 'url', or how can you base on?

**Reply by larsjanssen6** [reply]
For example in my pages I have directory structure `pages/app-1/index.vue`, `pages/app-2/index.vue`. I want domain1.com to serve `app-1/index.vue`, and domain2.com serve `app-2/index.vue`.

**Reply by xjccc** [reply]
I have a scene like this. But i create a monorepo project. Some common in my project/layers, and deploy site by itself.
